stock_internal_transfer/stock_internal_transfer.py

```diff
@@ -0,0 +1,171 @@
+        'mode': fields.selection(
+            lambda self, *a, **kw:
+            self._get_transfer_mode_sel(*a, **kw), string="Transfer Mode", ),
+    }
+
+    _defaults = {
+        'mode': 'default',
+    }
+
+    def onchange_location_dest_id(self, cr, uid, ids, location_dest_id,
+                                  context=None):
+        if location_dest_id:
+            location = self.pool['stock.location'].\
+                browse(cr, uid, location_dest_id, context=context)
+            return {'value': {'delivery_partner_id': location.partner_id.id}}
+        return {}
+
+    def _prepare_picking(self, cr, uid, ids, context=None):
+        assert len(ids) == 1, \
+            'This function should only be used for a single id at a time.'
+
+        wiz = self.browse(cr, uid, ids[0], context=context)
+        picking_data = {
+            'type': 'internal',
+            'origin': _('Internal Transfer Wizard'),
+            'location_id': wiz.location_id.id,
+            'location_dest_id': wiz.location_dest_id.id,
+            'partner_id': wiz.delivery_partner_id.id,
+            # we require all products to be available for shipment
+            'move_type': 'one',
+        }
+        return picking_data
+
+    def _create_move_lines(self, cr, uid, ids, picking_id, context=None):
+        assert len(ids) == 1, \
+            'This function should only be used for a single id at a time.'
+        move_obj = self.pool['stock.move']
+        picking_model = self.pool['stock.picking']
+
+        wizard = self.browse(cr, uid, ids[0], context=context)
+        picking = picking_model.browse(cr, uid, picking_id, context=context)
+
+        for line in wizard.line_ids:
+            # todo move to _prepare function
+            move_data = {
+                'product_id': line.product_id.id,
+                'product_qty': line.quantity,
+                'product_uom': line.product_uom.id,
+                'picking_id': picking.id,
+                'location_id': picking.location_id.id,
+                'location_dest_id': picking.location_dest_id.id,
+                'name': line.product_id.name_get()[0][1],
+            }
+            move_obj.create(cr, uid, move_data, context=context)
+
+    def _create_transfer(self, cr, uid, ids, context=None):
+        assert len(ids) == 1, \
+            'This function should only be used for a single id at a time.'
+        picking_obj = self.pool['stock.picking']
+
+        picking_data = self._prepare_picking(cr, uid, ids, context=context)
+        picking_id = picking_obj.create(cr, uid, picking_data, context=context)
+        self._create_move_lines(cr, uid, ids, picking_id, context=context)
+
+        wf_service = LocalService("workflow")
+        wf_service.trg_validate(uid, 'stock.picking',
+                                picking_id, 'button_confirm', cr)
+        picking_id = [picking_id]
+        picking_obj.action_assign(cr, uid, picking_id, context=context)
```

Comment:
I get this error message : TypeError: action_assign() got an unexpected keyword argument 'context'
And if I look action_assign, it seems the definition is : https://github.com/odoo/odoo/blob/7.0/addons/stock/stock.py#L768 ->  def action_assign(self, cr, uid, ids, *args)

so I think it is not possible to set **kwargs.